POC/api/main.py

```diff
@@ -186,17 +451,184 @@ async def create_database_field(
+    response_model=FieldInfo,
+    tags=["Field"],
+)
+async def api_delete_field(database_id: int, field_id: int):
+    with Session(db_engine) as session:
+        statement = select(FieldInfo).where(
+            FieldInfo.id == field_id and FieldInfo.db_id == database_id
+        )
+        db: FieldInfo | None = session.exec(statement).first()
+        if db is None:
+            raise HTTPException(status_code=404, detail="Field not found")
+        db.is_active = False
+        session.add(db)
+        session.commit()
+        session.refresh(db)
+    return db
+
+
+@app.delete(
+    "/api/backend/delete/databaseField/{database_id}",
+    response_model=Sequence[FieldInfo],
+    tags=["Field"],
+)
+async def api_delete_fields(database_id: int) -> Sequence[FieldInfo]:
+    with Session(db_engine) as session:
+        db_statement = select(DbInfo).where(DbInfo.id == database_id)
+        db: DbInfo | None = session.exec(db_statement).first()
+        if db is None or db.status == "deleted":
+            statement = select(FieldInfo).where(
+                FieldInfo.db_id == database_id and FieldInfo.is_active
+            )
+            db_fields: Sequence[FieldInfo] = session.exec(statement).all()
+            if db_fields is not None and len(db_fields) > 0:
+                for field in db_fields:
+                    field.is_active = False
+                session.add_all(db_fields)
+                session.commit()
+    return db_fields
+
+
+### DEFAULT ENDPOINTS AND REDIRECTS
+@app.get(
+    "/api/welcome",
+    response_model=FastUI,
+    response_model_exclude_none=True,
+    include_in_schema=False,
+)
+async def welcome():
+    form_routes: list[Route] = [
+        route
+        for route in app.routes
+        if isinstance(route, Route)
+        and "forms" in route.path
+        and "{" not in route.path
+        and route.methods is not None
+        and "GET" in route.methods
+    ]
+
+    return [
+        c.Page(
+            components=[
+                c.Div(
+                    components=[
+                        c.Heading(
+                            text=" ",
+                        ),
+                        c.Button(
+                            text=route.name,
+                            on_click=GoToEvent(url=route.path.replace("/api", "")),
+                        ),
+                    ]
+                )
+                for route in form_routes
+            ]
+        )
+    ]
+
+
+@app.get("/api/", response_class=RedirectResponse, tags=["Redirects"])
 async def home_redirect_backend() -> RedirectResponse:
     return RedirectResponse("/openapi.json", status_code=302)
 
 
-@app.get("/", response_class=RedirectResponse)
+@app.get("/", response_class=RedirectResponse, tags=["Redirects"])
 async def home_redirect() -> RedirectResponse:
-    return RedirectResponse("/docs", status_code=302)
+    return RedirectResponse("/welcome", status_code=302)
 
```

Comment:
More than 600 lines of code 🙈 

You probably want to leverage routers to distribute the code and make the main file more high level.

See: https://fastapi.tiangolo.com/tutorial/bigger-applications/

I suggest to refactor the code in a follow-up issue (on a new feature branch with a new PR).

Reply:
Sounds like a plan! Just trying to get this POC completed so i can begin to work on the MVP.

Reply:
see https://github.com/Tanner-Ray-Martin/Dynamic-DB/issues/33

Reply:
Just created a follow-up issue via `Reference in new issue`:
![image](https://github.com/user-attachments/assets/ef61371a-6511-4a17-9c80-aca0af972715)